layers/README.md

```diff
@@ -0,0 +1,85 @@
+
+| script | description |
+|:-------|:------------|
+| [exceptions.py](core/exceptions.py) | Implements errors and supporting functions for the core module. |
+| [filter.py](core/filter.py) | Implements a basic filter object. |
+| [gradient.py](core/gradient.py) | Implements a basic gradient object. |
+| [layer.py](core/layer.py) | Provides an interface for interacting with core module's layer representation. A further breakdown can be found in the corresponding section below. |
+| [layerobj.py](core/layerobj.py) | Implements a basic layer object. This should not be manipulated directly, instead, use the [layer representation](core/layer.py). |
+| [layout.py](core/layout.py) | Implements a basic layout object. |
+| [legenditem.py](core/legenditem.py) | Implements a basic legenditem object. |
+| [metadata.py](core/metadata.py) | Implements a basic metadata object. |
+| [technique.py](core/technique.py) | Implements a basic technique object. |
+| [layerops.py](manipulators/layerops.py) | Provides a means by which to combine multiple ATT&CK layer objects in customized ways. A further breakdown can be found in the corresponding section below. |
+
+## layer.py
+Layer.py provides a Layer class, which is the interface used to interact with the rest of the core module, and provides an abstract representation of an ATT&CK layer. The layer class has a collection of methods, the functionality of which is documented in the table below. An example of how to interface with the class to load and retrieve layer objects is documented following that.
+
+| method | description |
+|:-------|:------------|
+| Layer().load_input(_input_) | Loads an ATT&CK layer from either a dictionary or a string representation of a dictionary. |
+| Layer().load_file(_input_) | Loads an ATT&CK layer from a file location specified by the _input_. |
+| Layer().export_file(_input_) | Saves the current state of the loaded ATT&CK layer to a json file denoted by the _input_. |
+| Layer().get_dict() | Returns a representation of the current ATT&CK layer object as a dictionary. | 
+
+#### Example Usage
+
+```python
+example_layer_dict = {
+    "name": "example layer",
+    "version": "3.0",
+    "domain": "mitre-enterprise",
+    "description": "hello, world",
+    ...
+    }
+example_layer_location = "/path/to/layer/file.json"
+example_layer_out_location = "/path/to/new/layer/file.json"
+
+from layers.core.layer import Layer
+
+layer1 = Layer(example_layer_dict)
+layer1.export_file(example_layer_out_location)
+
+layer2 = Layer()
+layer2.load_input(example_layer_dict)
+layer2.get_dict()
+
+layer3 = Layer()
+layer3.load_file(example_layer_location)
```

Comment:
Should add python comments explaining what each step is doing. E.g `# load the layer dict into a Layer instance`, `# Initialize a layer based on the contents of a file`. Even though the code is pretty self explanatory people look to comments first for the purposes of organizing blocks of code in examples.

Also, `layer2` doesn't really need to exist, right? You could basically call `data = layer1.get_dict()` instead of `layer2.get_dict()` since layer2 and layer1 are identical. Doing it this way seems to imply that a layer is destroyed after exporting.

Reply:
Also, probably good to assign the result of `get_dict()` to some variable since otherwise there's no point to that function call :)

Reply:
I'm not following that creating a layer2 implies layer1 was destroyed after exporting. The reason layer2's example exists is because layer2 is demonstrating that you can load data after initialization, as layer1 loaded it during initialization. Adding documentation is a good idea, though, so I've done that.

Reply:
Oh I see, I missed the `load_input` part. With that in mind it's fine to keep layer2 and layer1 separated.